bagit.py

```diff
@@ -1097,13 +1097,12 @@ def _make_tag_file(bag_info_path, bag_info):
     headers = sorted(bag_info.keys())
     with open_text_file(bag_info_path, 'w') as f:
         for h in headers:
-            if isinstance(bag_info[h], list):
-                for val in bag_info[h]:
-                    f.write("%s: %s\n" % (h, val))
-            else:
-                txt = bag_info[h]
+            values = bag_info[h]
+            if not isinstance(values, list):
+                values = [values]
+            for txt in values:
                 # strip CR, LF and CRLF so they don't mess up the tag file
-                txt = re.sub(r'\n|\r|(\r\n)', '', txt)
+                txt = re.sub(r'\n|\r|(\r\n)', '', "%s" % txt)
```

Comment:
Nope, there's a test case using a list of ints, which passed until now because the substitution wasn't applied to lists - and substitution on an int fails. And we can't guarantee API users will always pass strings.
I chose this approach because it works for both Python 2 and 3 and uses the correct string class on both.